base/array.jl

```diff
@@ -827,32 +827,36 @@ function findmax(a)
     if isempty(a)
         throw(ArgumentError("collection must be non-empty"))
     end
-    m = a[1]
-    mi = 1
-    for i=2:length(a)
-        ai = a[i]
+    i = start(a)
+    mi = i
+    m, i = next(a, i)
+    while !done(a, i)
+        iold = i
+        ai, i = next(a, i)
         if ai > m || m!=m
             m = ai
-            mi = i
+            mi = iold
         end
     end
-    return (m, mi)
+    return (m, iterstate(mi))
```

Comment:
Heck, `sub2ind` is fairly fast.  `iterstate(x::CartesianIndex) = ind2sub(X)`?  It'll probably still be a net gain for linear slow arrays.